File: python/evals/autocontext_fixture/src/api/router.py
"""APIRouter — imports auth middleware, registers endpoints."""
from src.auth.middleware import auth_middleware
from src.auth.login import handle_login
from src.api.rate_limit import rate_limit_decorator


class APIRouter:
    def __init__(self):
        self.routes = {}

    def add(self, path, handler):
        self.routes[path] = handler

    def dispatch(self, request):
        err = auth_middleware(request)
        if err:
            return err
        handler = self.routes.get(request.path)
        if not handler:
            return {"error": "not found"}, 404
        return handler(request)


def build_router():
    router = APIRouter()
    router.add("/login", rate_limit_decorator(handle_login))
    return router

File: python/evals/autocontext_fixture_js/src/server/router.ts
// Router with route table + dispatch.
import { handleLogin } from "../auth/login";
import { authMiddleware } from "../auth/middleware";
import { rateLimit } from "../api/rateLimit";

export type Handler = (req: any) => [any, number];

export function createRouter() {
  const routes: Record<string, Handler> = {};
  return {
    add(path: string, handler: Handler) { routes[path] = handler; },
    dispatch(req: any) {
      const err = authMiddleware(req);
      if (err) return err;
      const h = routes[req.path];
      if (!h) return [{ error: "not found" }, 404];
      return h(req);
    },
    register() {
      this.add("/login", rateLimit(handleLogin, 5));
    },
  };
}
